pkg/storage/secret/metadata/secure_value_store.go

```diff
@@ -353,11 +355,12 @@ func (s *secureValueMetadataStorage) ReadForDecrypt(ctx context.Context, namespa
 	return secureValue, nil
 }
 
-func (s *secureValueMetadataStorage) read(ctx context.Context, namespace xkube.Namespace, name string) (secureValueDB, error) {
+func (s *secureValueMetadataStorage) read(ctx context.Context, namespace xkube.Namespace, name string, isForUpdate sqlForUpdate) (secureValueDB, error) {
 	req := readSecureValue{
 		SQLTemplate: sqltemplate.New(s.dialect),
 		Namespace:   namespace.String(),
 		Name:        name,
+		IsForUpdate: isForUpdate, // TODO: make this smarter, check if there is a tx in the context and set to TRUE automatically?.
```

Comment:
@PoorlyDefinedBehaviour what do you think about this? can add a method to check if there's already a transaction in the context, and automatically make this `true` 🤔 (would we ever want this to be `false` inside transactions due to consistency?)